### DISCUSSION TITLE
Line Spacing

### DISCUSSION THREAD

**Discussion by writingpoli** [knowledgefutures/pubpub#2298] [Support]
A problem we continually run into (especially with poetry submissions) is the auto-double-spacing of each line whether imported or copy-and-pasted into the editor. Is there a way we could default to single spacing? Or maintaining original spacing from imported files? 

We're generally forced to backspace each line in a stanza, hit SHIFT+ENTER to get single-spaced lines correct. Then just ENTER for a stanza break. 

This makes the reviewing and editing process much more difficult because stanzas are often not separated -- instead, everything just comes out double-spaced upon first read, which can significantly change the flow/meaning of a work. And authors tend not to understand how to fix the spacing themselves, so it adds to the editing/reviewing workload. 

Thoughts? Ideas? Suggestions? 

Thank you!

**Comment by gabestein** [comment]
Can you send an example of a word file to help@? If folks are using carriage returns in Word for new lines and setting margins to be smaller, that may be tough for us. But if for some reason we’re not picking up soft vs hard breaks in word there may be a fix we can do.

**Reply by writingpoli** [reply]
I think poets tend to just hit ENTER to go to next line and then two ENTERs to break a stanza, so that's probably going to cause y'all issues. I'll have our new poetry editor send over an example for you all to play with, though. Thank you for the weekend response! :)

**Reply by gabestein** [reply]
Gotcha. Yeah, that’s a tough one. Have you tried using custom css to play with paragraph margins?

**Reply by writingpoli** [reply]
Oh, boy, now you're asking a lot of me. Pretty sure if I start playing with the CSS, I'll break everything -- including our prose and artwork Pubs. I will keep that in my back pocket, though.

**Reply by freeversefree** [reply]
Hi, Gabe. This is Arvilla, the new poetry editor for SAR. I will attach a couple of the documents sent by our submitters.Poetry is almost always single spaced with specific stanza breaks. We would like to be able to retain the original format in PubPub instead of it defaulting to double-spacing. The double-spacing occurs whether authors send a pdf or doc document. 

[7,7 Poem.pdf](https://github.com/pubpub/pubpub/files/9852446/7.7.Poem.pdf)
[Long Lost.docx](https://github.com/pubpub/pubpub/files/9852468/Long.Lost.docx)

Part of the issue could be on the author's end, but it does seem that their lines are set to "single space." So, I'm not sure how we can drop them in PubPub and keep the single spacing with stanza breaks.

**Reply by gabestein** [reply]
Thanks for the examples! Taking a look at the Word doc (PDFs are another thing entirely and notoriously finnicky — there's not much we can do there, sadly), I do think the issue here is a mismatch between Word formatting and HTML structure, and I'm not sure there's a great way to fix it automatically.

Long story short, pressing enter in Word results in a paragraph break, which when converted to HTML, is read as a new paragraph. The margins PubPub puts around paragraphs are what create the appearance of a double space, but in reality it is correctly reading those as what they are in both Word's internal structure and HTML: new paragraphs. 

Ideally, authors would use [hard breaks](https://practicaltypography.com/hard-line-breaks.html) (shift+enter) for lines and paragraph breaks (just enter) for stanzas. But I know that changing that behavior might be a tall order. One fix could be to target paragraph margins for poetry pieces in custom CSS, which is something I can help with (although it looks like it'll require a small code change on our end). You could also use find/replace to convert those paragraph breaks into hard breaks in bulk before import. Or you could ask authors to submit with hard breaks rather than paragraph breaks.

**Reply by freeversefree** [reply]
Thank you, Gabe.
Could you possibly make the small code change on your end, and we will see where that lands.

We can definitely put in our submission instructions for poetry writers to use the hard breaks, but like you said, old habits die hard. It would be a learning curve, for sure.

Thanks for your help and input in this mattter.
Arvilla

**Reply by writingpoli** [reply]
Thanks so much @gabestein. I'll definitely need your assistance on inputting the correct CSS so it doesn't break everything else in our community, too. (At least once it's all said and done, though, we can direct folks to this discussion for the answer!) 

😃
